app/models/users/user.rb

```diff
@@ -5,49 +5,69 @@ class User < ActiveRecord::Base
   belongs_to :grove
 
   def self.from_omniauth(access_token)
-    data = access_token.info
-    if within_roots?(access_token)
-      if Teacher.is_teacher?(data)
-        user = Teacher.where(email: data["email"]).first
-        user = check_or_create_token(user, access_token)
-        user = check_or_create_image(user, data['image'])
-      elsif Student.is_student?(data)
-        user = Student.where(email: data["email"]).first
-        user = check_or_create_token(user, access_token)
-        user = check_or_create_image(user, data['image'])      
-      end
-      user
-    end
-  end
-
-  def self.within_roots?(data)
-    data["extra"]["id_info"]["hd"] == "rootselementary.org"
-  end
-
-  def self.check_or_create_token(user, access_token)
-    unless user.refresh_token
-      creds = access_token["credentials"]
-      user.update_attributes(token: creds["token"],
-                             refresh_token: creds["refresh_token"],
-                             expires_at: creds["expires_at"])
+    package = OmniauthPackage.new(access_token)
+    user = User.where(email: package.email).first
```

Comment:
@NickyBobby I just wanted you to double check my logic here. Since we are adding the users to the system, I'm not really sure we need to check the domain. I added that accessor to the `OmniauthPackage` class but am not using it. User is either in the database or not. Am I missing something here. I think originally it would have been so that we could find _or create_ a user based on domain but I dont  think thats the case anymore.  LIkewise, at this point for this request, a user is a user, so no need to switch on type.